graphene/src/layers/layer_info.rs

```diff
@@ -86,11 +89,11 @@ pub trait LayerData {
 	/// let quad = Quad::from_box([DVec2::ZERO, DVec2::ONE]);
 	/// let mut intersections = vec![];
 	///
-	/// shape.intersects_quad(quad, &mut vec![shape_id], &mut intersections);
+	/// shape.intersects_quad(quad, &mut vec![shape_id], &mut intersections, &Default::default());
 	///
 	/// assert_eq!(intersections, vec![vec![shape_id]]);
 	/// ```
-	fn intersects_quad(&self, quad: Quad, path: &mut Vec<LayerId>, intersections: &mut Vec<Vec<LayerId>>);
+	fn intersects_quad(&self, quad: Quad, path: &mut Vec<LayerId>, intersections: &mut Vec<Vec<LayerId>>, font_cache: &FontCache);
```

Comment:
For this and all these other functions related to transformation, it seems pretty inappropriate to have to pass the font cache here. Can the cache instead live as a reference on `struct Layer`? Presumably that means lifetimes may get tricky, or perhaps that's even illogical to make Graphene handle that. Maybe give a copy of the data to Graphene? I'm not really sure what the best solution is here. @TrueDoctor perhaps you have any thoughts?

Reply:
@Keavon Don't think so (unless we use something from std::sync), as we need to get mutable access to the font cache (i.e. dropping all other references) when we load a new font.